src/modules/mavlink/mavlink_main.cpp

```diff
@@ -243,7 +249,7 @@ Mavlink::Mavlink() :
 	_rate_tx(0.0f),
 	_rate_txerr(0.0f),
 	_rate_rx(0.0f),
-#ifdef __PX4_POSIX
+#if defined(CONFIG_NET) || defined(__PX4_POSIX)
```

Comment:
Creating a new define `__PX4_NETWORK` and setting that above for when NuttX networking is enabled or POSIX is available is cleaner.